### PR TITLE
Data visualization in R blog post -- Caroline Riedstra

### PR DESCRIPTION
**Pull request by cpriedstra**
This post outlines the basics of generating figures in R using ggplot and tidyverse. It includes an introduction to different graphing basics and features of proper figure generation.

### CODE REVIEW

**Comment by cpriedstra** [comment]
Hi @claudiaflautero and @naupaka , thank you for the feedback! I've added a few script examples for the different styles of plots as recommended by @claudiaflautero. Great idea! Please let me know if there is anything else either of you suggest!

File: _posts/2020-11-05-data-visualization-in-R.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+# Using  `ggplot`  
+`ggplot` is a data visualization system from tidyverse that takes input data sets and creates customizable graphics. For information beyond this post and user guides, github provides a [ggplot overview](https://ggplot2.tidyverse.org/index.html). Prior to running any script you will have to load the library for  `ggplot` or `ggplot2`.  Note that this is different than installing the packages.
+
+To do this: ```library(ggplot2)```
+
+## Writing a script and data input
+The general format of a ggplot input is as follows:
+```
+{r name_of_plot}
```
Comment (naupaka) [inline _posts/2020-11-05-data-visualization-in-R.md]: Don't need this actually -- this is just for Rmarkdown chunks, so I would remove it.

File: _posts/2020-11-05-data-visualization-in-R.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+                         "file_name.png"))
+```
+
+Finally, in R, if you want to see the generated figure without opening the saved file, you just call the plot name at the end of the script:
+```
+name_of_plot
+```
+
+Interpreting the script above based on inputs and commands:  
+1. `{r name_of_plot}`
```
Comment (naupaka) [inline _posts/2020-11-05-data-visualization-in-R.md]: I'd remove it here too and just renumber your list to start with 1 at 'data'

File: _posts/2020-11-05-data-visualization-in-R.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+1. `{r name_of_plot}`  
+  * In this command we are calling for R to create a plot with this specific title
+2. `data = `
+  * We have to read in the data file, the easiest way to do this is from a csv or txt file and prior to creating the figure set a variable equal to this file. This way you can call the variable and that will read in the data as opposed to giving the precise location of the data each time.
+3. `aes`
+  * This is where you specify how data is being plotted. Which data is going on the x, which data is going on the y. You can also opt to select for color, size, and and data point shape or transparency (depending what type of plot you choose these options differ slightly).
+    * Within `aes` you a can also add a fill. You can use this when you are comparing two independent variable that have been grouped:
+      * `ggplot(aes(x = city, y = count, fill = type))`
+    * Alternatively, you can also choose to identify different groups by using multiple colors based on the group ID:
+      * ``
```
Comment (naupaka) [inline _posts/2020-11-05-data-visualization-in-R.md]: Not sure if there was supposed to be something here?

File: _posts/2020-11-05-data-visualization-in-R.md [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+Finally, in R, if you want to see the generated figure without opening the saved file, you just call the plot name at the end of the script:
+```
+name_of_plot
+```
+
+Interpreting the script above based on inputs and commands:  
+1. `{r name_of_plot}`  
+  * In this command we are calling for R to create a plot with this specific title
+2. `data = `
+  * We have to read in the data file, the easiest way to do this is from a csv or txt file and prior to creating the figure set a variable equal to this file. This way you can call the variable and that will read in the data as opposed to giving the precise location of the data each time.
```
Comment (naupaka) [inline _posts/2020-11-05-data-visualization-in-R.md]: Not exactly. When the command is run to create the data object, it reads from the file and stores the data into memory. That happens when the variable is created, not when the ggplot call occurs. So maybe a slight rewording here would be good.

File: _posts/2020-11-05-data-visualization-in-R.md [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+        `ggplot(aes(x = independent_variable,
+        y = dependent_variable)) +
+        geom_line() +
+        labs(title = "Title",
+        x = "Independent Variable",
+        y = "Dependent Variable")`
+5. `position =`
+  * In this sample script, `position_dodge` was used to group the bar plot by category. Depending on your ultimate figure goal, this may not be necessary for all bar plots. Occasionally a stacked position, the default, is better when using a grouped dataset.
+6. `labs`
+  * Axis labels, can be set `title`, `x=`, and `y=`
```
Comment (naupaka) [inline _posts/2020-11-05-data-visualization-in-R.md]: Add spaces around the `=`

File: _posts/2020-11-05-data-visualization-in-R.md [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+7. In `ggplot` to continue adding input, each section has to be connection with a '+'.
+  * Other input options could be the following:
+    * `scale_fill_gradient(low = color1", high = "color2")`: color 1 and color 2 are up to you to choose with in a ggplot color scheme. This function can be used when you selected a 'fill' option for grouped variables.
```
Comment (naupaka) [inline _posts/2020-11-05-data-visualization-in-R.md]: Looks like you're missing an open quotes on color1

File: _posts/2020-11-05-data-visualization-in-R.md [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+    * `theme_few()` will remove the gray background from plots as well as the grid
```
Comment (naupaka) [inline _posts/2020-11-05-data-visualization-in-R.md]: Might want to point out that this is in `ggthemes` and isn't in `ggplot2` by default.

**Review by naupaka — CHANGES_REQUESTED** [review]
Nice job! Some small suggestions to improve some points to be more accurate, and a couple typos. @ mention me when you've made the changes and I'll take another look so we can merge it in.

**Comment by naupaka** [comment]
Also! Sorry this took me so long to get to.

**Comment by cpriedstra** [comment]
Hi @naupaka, thank you for the feedback! I've made the requested changes: 8cc771437bd656b515c93efa71242ec23befdd16

File: _posts/2020-11-05-data-visualization-in-R.md [hunk 8]
```diff
@@ -72,16 +71,16 @@ name_of_plot
 ```
 
 Interpreting the script above based on inputs and commands:  
-1. `{r name_of_plot}`  
-  * In this command we are calling for R to create a plot with this specific title
-2. `data = `
-  * We have to read in the data file, the easiest way to do this is from a csv or txt file and prior to creating the figure set a variable equal to this file. This way you can call the variable and that will read in the data as opposed to giving the precise location of the data each time.
+1. `data = `
+  * Here we are calling the dataset we want to use for the plot. The data set should be a csv or txt file that has already been read into the script and is called as a variable in this section of the plotting code.
 3. `aes`
```
Comment (naupaka) [inline _posts/2020-11-05-data-visualization-in-R.md]: Should fix all the numbering. Currently this skips 2 and jumps to 3.

File: _posts/2020-11-05-data-visualization-in-R.md [hunk 9]
```diff
@@ -114,12 +113,13 @@ Interpreting the script above based on inputs and commands:
...
 6. `labs`
-  * Axis labels, can be set `title`, `x=`, and `y=`
+  * Axis labels, can be set `title`, `x =`, and `y =`
 7. In `ggplot` to continue adding input, each section has to be connection with a '+'.
   * Other input options could be the following:
-    * `scale_fill_gradient(low = color1", high = "color2")`: color 1 and color 2 are up to you to choose with in a ggplot color scheme. This function can be used when you selected a 'fill' option for grouped variables.
-    * `theme_few()` will remove the gray background from plots as well as the grid
-    * `theme(axis.text.x = element_text(angle = 0-180))` will rotate the text on the x axis between 0 and 180 degrees. The same can be done for values on the y axis.
+    * `scale_fill_gradient(low = "color1", high = "color2")`: color 1 and color 2 are up to you to choose with in a ggplot color scheme. This function can be used when you selected a 'fill' option for grouped variables.
+    * `ggthemes` can be used as well for additional customization however it is not part of `ggplot2` and needs to be loaded separately with `library(ggthemes)`:
```
Comment (naupaka) [inline _posts/2020-11-05-data-visualization-in-R.md]: I'd say `ggthemes` is a package that...

**Review by naupaka — CHANGES_REQUESTED** [review]
Couple more minor tweaks to make, and then this should be good to go.

**Comment by cpriedstra** [comment]
Whoops, great catch, thank you! @naupaka 0321380f11fa01f66bb9001f191be8e28380cdde

**Review by naupaka — APPROVED** [review]

**Comment by naupaka** [comment]
👍

**Comment by naupaka** [comment]
https://2020f-usfca-btec-640.github.io/ggplot/figures/r-formatting/2020/11/07/data-visualization-in-R.html

**Pull request merged**